src/components/App/App.jsx

```diff
@@ -0,0 +1,412 @@
+            title: post.attributes.title,
+            url: post.attributes.url,
+            is_public: post.attributes.is_public,
+          }));
+          const sortedPosts = extractedPosts.sort((a, b) => {
+            return b.is_public - a.is_public;
+          });
+
+          setPatreonPosts(sortedPosts);
+        })
+        .catch((error) => {
+          console.error("Error getting Patreon posts:", error);
+        });
+    };
+
+    return handleSubmit(requestGetPatreonPosts);
+  };
+
+  const handleGetBookPages = () => {
+    const requestGetBookPages = () => {
+      return getPages()
+        .then((imageUrls) => {
+          if (imageUrls && imageUrls.length > 0) {
+            setBookPages(imageUrls);
+          }
+        })
+        .catch((error) => {
+          console.error("Error getting Patreon posts:", error);
+        });
+    };
+
+    return handleSubmit(requestGetBookPages);
+  };
+
+  //// HANDLE BOOKS ////
+
+  const handlePageChange = (newPageIndex) => {
+    setUserBookIndex(newPageIndex);
+    if (isLoggedIn) {
+      return handleEditProfile({ page: newPageIndex });
+    }
+  };
+
+  //// MODALS ////
+  const openModals = (evt) => {
+    setActiveModal(evt.target.id);
+  };
+
+  const closeModals = () => {
+    setActiveModal(null);
+    if (commentToEdit) {
+      setCommentToEdit(null);
+    }
+
+    // set timeout so card wont visually change before the modal fades away
+    setTimeout(() => {}, 400);
+  };
+
+  // sets/removes event listeners whenever a modal is opened/closed
+  React.useEffect(() => {
+    const handleOutsideClick = (evt) => {
+      if (evt.target.id === activeModal) {
+        closeModals();
+      }
+    };
+    const handleEscClose = (evt) => {
+      if (evt.key === "Escape") {
+        closeModals();
+      }
+    };
+
+    if (activeModal) {
+      document.addEventListener("mousedown", handleOutsideClick);
+      document.addEventListener("keydown", handleEscClose);
+      return () => {
+        document.removeEventListener("mousedown", handleOutsideClick);
+        document.removeEventListener("keydown", handleEscClose);
+      };
+    }
+  }, [activeModal, closeModals]);
+
+  //// USE EFFECTS ////
+
+  // Check if user has token //
+  React.useEffect(() => {
+    const token = localStorage.getItem("jwt");
+    if (token) {
+      authorizeToken(token)
+        .then((res) => {
+          setCurrentUser(res.data);
+          setUserBookIndex(res.data.page);
+          setUserToken(token);
+          setIsLoggedIn(true);
+        })
+        .catch(console.error);
+    }
+  }, []);
+
+  React.useEffect(() => {
+    if (isLoggedIn) handleGetUserComments();
+  }, [isLoggedIn]);
+
+  React.useEffect(() => {
+    handleGetPatreonPosts();
+  }, []);
+
+  React.useEffect(() => {
+    handleGetBookPages();
+  }, []);
+
+  React.useEffect(() => {
+    handleGetPageComments(userBookIndex + 1);
+  }, [userBookIndex]);
+
+  return (
+    <>
```

Comment:
`<>` empty tags are not needed if you have a common tag that wraps the whole markup of the component